actiontext/test/unit/content_test.rb

```diff
@@ -134,6 +134,33 @@ class ActionText::ContentTest < ActiveSupport::TestCase
     assert_match %r/\A#{Regexp.escape '<div class="trix-content">'}/, rendered
   end
 
+  test "replace certain nodes" do
+    html = <<~HTML
+      <div>
+        <p>replace me</p>
+        <p>ignore me</p>
+      </div>
+    HTML
+
+    expected_html = <<~HTML
+      <div>
+        <p>replaced</p>
+        <p>ignore me</p>
+      </div>
+    HTML
+
+    content = content_from_html(html)
+    replaced_fragment = content.fragment.replace("p") do |node|
+      if node.text =~ /replace me/
+        node.inner_html = "<p>replaced</p>"
+      else
+        node
+      end
```

Comment:
Setting `node.inner_html = "<p>replaced</p>"` will cause the node to be `<p><p>replaced</p></p>`.  The reason this test passes, though, is because `#inner_html=` returns the input string (as most setters would), so the block just returns the string `"<p>replaced</p>"`.

So this should probably be:

```suggestion
      if node.text =~ /replace me/
        "<p>replaced</p>"
      else
        node
      end
```

Reply:
Totally @jonathanhefner, my mistake 🙏 .